src/userList2.js

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import UserList2ClientBreakdown from "./userList2ClientBreakdown";
+
+export default function UserList2(props) {
+
+    const userDataList = props.userDataList;
+
+    // create a distinct list of clients
+    let clientList = [];
+    userDataList.map(userData => {
+        clientList.push(userData.client)
+    });
+    clientList = clientList.filter((value, index, self) => {
+        return self.indexOf(value) === index;
+    })
+
+    return <div> 
+        <h1>User List 2 - Grouped By Client / Vendor  </h1>
+        <ul>
+            { clientList.map(client => 
+                <UserList2ClientBreakdown client={client} userDataList={userDataList} />
+            )}           
+        </ul>
+        
+    </div>
```

Comment:
Just some minor spacing/formatting here. I recommend getting the VS Code extension "Prettier - Code formatter" as it helps with this kind of thing and keeps everyone consistent. Just need to go to VS Code's Extensions tab and download/install it, and then modify the settings through the UI or JSON (ctrl + shift + p, search 'settings') and make changes to these settings:
```
"editor.defaultFormatter": "esbenp.prettier-vscode",
"editor.formatOnSave": true
```
```suggestion
   return (
    <div>
      <h1>User List 2 - Grouped By Client / Vendor </h1>
      <ul>
        {clientList.map((client) => (
          <UserList2ClientBreakdown
            client={client}
            userDataList={userDataList}
          />
        ))}
      </ul>
    </div>
  );
```

Reply:
Awesome thanks!  It's like Ctrl-Alt-F in Resharper